src/pybind/mgr/dashboard/frontend/e2e/rgw/buckets.po.ts

```diff
@@ -0,0 +1,10 @@
+import { browser } from 'protractor';
+
+export class BucketsPage {
+  navigateTo() {
+    return browser.get('/#/rgw/bucket');
+  }
+  wait() {
+    return browser.sleep(1000);
```

Comment:
I would recommend we use `browser.wait(until...` instead of a fixed sleep.
Please have a look at https://github.com/openattic/openattic/blob/master/webui/e2e/common.js#L251
We could add those method in `helper.po.ts`.

Reply:
I implemented it as so 
```
static getBreadcrumbText() {
  const breadcrumb = $('.breadcrumb-item.active');
  browser.wait(
    Helper.EC.presenceOf(breadcrumb),
    Helper.TIMEOUT,
    'Element taking too long to appear in the DOM'
  );
  return breadcrumb.getText();
}
```
Unfortunately this method causes the tests to fail 
```
**************************************************
*                    Failures                    *
**************************************************

1) RGW buckets page should open and show breadcrumb
  - Expected 'Pools' to equal 'Buckets'.

2) RGW daemons page should open and show breadcrumb
  - Expected 'Pools' to equal 'Daemons'.

3) RGW users page should open and show breadcrumb
  - Expected 'Pools' to equal 'Users'.
```
The sleeping implementation was previously addressed here: https://github.com/ceph/ceph/pull/25451#issuecomment-447400893

Reply:
Another option is to implement the navigateTo() functions as follows, with the sleep implicitly included to allow Selenium to fully load the page before proceeding 

```
export class Page {
  navigateTo() {
    browser.get('/#/page');
    browser.sleep(1000);
  }
}
```

Reply:
I just created a PR to try to fix a waiting problem during the authentication phase.
If that works, it might also remove the need to have waits in this PR.

https://github.com/ceph/ceph/pull/26384

Reply:
Looks like the previous PR didn't fix this problem.

This looks like its a race condition between the update of the breadcrumb and the load of the page component, and for some reason RGW is always winning :)

I have changed the breadcrumb component, so we can detect when it is changing pages:
https://github.com/tspmelo/ceph/commit/a47aab20625c009a944b5be1b5b5d5020550e696
with this you can remove all the sleeps and the e2e tests will work.

I think it will be quicker if you just cherrypick my commit into this PR.

Reply:
@tspmelo What would be the cleanest way to do the cherrypick? I will also take out the waits

Reply:
While you have your branch checked out,
I would suggest you to fetch my branch and then do a rebase on top of it with:
`git rebase -i tspmelo/wip-breadcrumb-pending`